src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorsAndIndexer/ConversionCompletionProvider.cs

```diff
@@ -0,0 +1,290 @@
+                           displayTextSuffix: $"{optionalNullableQuestionmark})",
+                           filterText: targetTypeName,
+                           sortText: SortText(targetTypeName),
+                           glyph: Glyph.Operator,
+                           rules: CompletionItemRules.Default,
+                           properties: CreatePropertiesBag(
+                               (MinimalTypeNamePropertyName, $"{targetTypeName}{optionalNullableQuestionmark}"),
+                               (ContainerTypeNamePropertyName, containerTypeName),
+                               ("ContextPosition", position.ToString())));
+        }
+
+        private static string GetOptionalNullableQuestionMark(bool isNullable)
+            => isNullable ? "?" : "";
+
+        protected override async Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+            => SymbolCompletionItem.HasSymbols(item)
+                ? await base.GetDescriptionWorkerAsync(document, item, cancellationToken).ConfigureAwait(false)
+                : GetBuildInConversionDescription(item);
+
+        private static CompletionDescription GetBuildInConversionDescription(CompletionItem item)
+        {
+            Contract.ThrowIfFalse(item.Properties.TryGetValue(MinimalTypeNamePropertyName, out var targetTypeName));
+            Contract.ThrowIfFalse(item.Properties.TryGetValue(ContainerTypeNamePropertyName, out var sourceTypeName));
+            var taggedText = new[]
+            {
+                new TaggedText(TextTags.Keyword, sourceTypeName),
+                new TaggedText(TextTags.Punctuation, "."),
+                new TaggedText(TextTags.Keyword, "explicit"),
+                new TaggedText(TextTags.Space, " "),
+                new TaggedText(TextTags.Keyword, "operator"),
+                new TaggedText(TextTags.Space, " "),
+                new TaggedText(TextTags.Keyword, targetTypeName),
+                new TaggedText(TextTags.Punctuation, "("),
+                new TaggedText(TextTags.Keyword, sourceTypeName),
+                new TaggedText(TextTags.Space, " "),
+                new TaggedText(TextTags.Parameter, "value"),
+                new TaggedText(TextTags.Punctuation, ")"),
+                new TaggedText(TextTags.LineBreak, Environment.NewLine),
+                new TaggedText(TextTags.Text, "Defines an explicit conversion of a"),
+                new TaggedText(TextTags.Space, " "),
+                new TaggedText(TextTags.Keyword, sourceTypeName),
+                new TaggedText(TextTags.Space, " "),
+                new TaggedText(TextTags.Text, "to a"),
+                new TaggedText(TextTags.Space, " "),
+                new TaggedText(TextTags.Keyword, targetTypeName),
+                new TaggedText(TextTags.Punctuation, "."),
+            };
+
+            return CompletionDescription.Create(taggedText.ToImmutableArray());
```

Comment:
The description is derived and altered from `decimal`
![decimal](https://user-images.githubusercontent.com/24472128/94853173-43a62980-042b-11eb-823d-b307acaa58db.png)

I also experimented with a class implementing `IMethodSymbol` and call `ISymbolDisplayService` and `IDocumentationCommentFormattingService` but that ended up being too involved. So I decided to copy the `TaggedText[]` of the decimal and use that as a starting point.

It lacks the following features:
* The classification of source and target type as `keyword` is wrong for `Enums`
* The description is a bit misleading. It describes the operator function, but it would be more useful to describe the conversion itself, like "Explicit cast from int to byte".
* The description should not contain *of a* and *to a* to avoid grammar issues with type names starting with a vowel.
* Needs to be localized.
* Is `Environment.NewLine` correct here?

Given, that the idea to create the `CompletionDescription` out of `TaggedText` is the right way to go here, I would like to change the description text and localize it.

Reply:
so localization is hte pri1 bit here.  Any solution you create must be totally localizable.  that means not piecing together strings in a piecemeal fashion.  if you can accomplish this, then this approach is ok.  If you cannot, then it won't be.

Reply:
I see two options:
```CS
FeaturesResources.BuiltInConversionDescription_1 = "Explicit cast from";
FeaturesResources.BuiltInConversionDescription_2 = "to";
```
or 
```CS
FeaturesResources.BuiltInConversionDescription = "Explicit cast from|to";
...
var descriptionParts = FeaturesResources.BuiltInConversionDescription.Split('|');
```
Would any of these two be acceptable?

Reply:
neither of htese work, because they presume that all all languages can form real sentences by taking two chunks of text and concatenating them in this fashion.  the only type of approach that can work is if there's a loc string of the form: `"foo bar {1} baz quux {0} ztesch"`  this lets the loc person determine precisely what string will be shown, with the placeholders for the two values to go into (potentially in different orders depending on language).